k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/GenericK8sResourceHandler.kt

```diff
@@ -65,6 +67,22 @@ abstract class GenericK8sResourceHandler <S: GenericK8sLocatable, R: K8sManifest
     // to the more generic Clouddriver function
     abstract suspend fun getK8sResource(r: Resource<S>) : R?
 
+    override suspend fun current(r: Resource<S>): R? =
+        getK8sResource(r)?.let{
+            log.debug("response from clouddriver for manifest: $it")
+            it.status?.let {
+                if (it.isReady()) {
+                    log.info("${r.spec.displayName} is healthy" )
+                    eventPublisher.publishEvent(ResourceHealthEvent(r, true))
+                } else {
+                    log.info("${r.spec.displayName} is not healthy")
+                    eventPublisher.publishEvent(ResourceHealthEvent(r, false))
+                    throw ResourceNotReady(r)
+                }
+            }
+            it
+        }
+
```

Comment:
We should ensure we return `ResourceNotReady` or null when clouddriver returns 404. Otherwise we are going to be filling keel log with 404 exceptions due to user error. Also I think it is possible for clouddriver to return 404 especially when it's still being provisioned.

Reply:
isnt that whats happening already?

Reply:
oh, did you mean either or by this `ensure we return ResourceNotReady or null when clouddriver returns 404`?

we are returning null, which cascades to `null` in the subclasses as well.

Reply:
So if there are any errors during `getK8sResource(r)` , it's going to throw an exception. It's not an issue if implementation of  `getK8sResource(r)` caught specific errors and returned null. I am not seeing that in any of implementations.

Reply:
yeah so 404 return null, anything else throws an exception which bubbles up and sets the status of the resource to `ERROR`. why do we want to catch the exception? The exception will go all the way to Keel for it to be caught and reported.